### DISCUSSION TITLE
Load thumbnail from EXIF

### DISCUSSION THREAD

**Discussion by adrium** [filebrowser/filebrowser#1188] [Ideas]
**Is your feature request related to a problem? Please describe.**
Filebrowser on Raspberry 4 is already quite performant. However, when load is high from other processes it feels quite slow when generating thumbnails.

**Describe the solution you'd like**
Thumbnails are often already present inside the EXIF of JPEG files.
They could be served directly to the client and a huge performance increase can be expected.

**Describe alternatives you've considered**
For files with an EXIF present, this would be one of the fastest approaches. Other software that generate thumbnails are usually slower.

**Comment by o1egl** [comment]
To keep it simple and unified for all the image types. I have some ideas to track file system changes and add some background job triggers. So thumbnails generation could be a part of it.

**Comment by adrium** [comment]
Hi o1egl, I understand your concern. I gave it a go and the result is impressive, please have a look at #1234.

**Comment by adrium** [comment]
Hi Oleg, I actually appreciate it much that filebrowser does not fill the disk with caching stuff.
I have been using the PR for almost a month without major issues. I added some small fixes just recently.
